pkg/validation/test.go

```diff
@@ -808,7 +808,10 @@ func validateCredentials(fieldRoot string, credentials []api.CredentialReference
 		for j, other := range credentials[i+1:] {
 			index := i + j + 1
 			if credential.MountPath == other.MountPath {
-				errs = append(errs, fmt.Errorf("%s.credentials[%d] and credentials[%d] mount to the same location (%s)", fieldRoot, i, index, credential.MountPath))
+				if credential.Collection == other.Collection && credential.Name != other.Name {
+					continue
+				}
+				errs = append(errs, fmt.Errorf("%s.credentials[%d] and credentials[%d] mount to the same location (%s), but are in different collections", fieldRoot, i, index, credential.MountPath))
```

Comment:
```yaml
credentials:
  - collection: collection-a
    name: a1
    mount_path: /mnt/secrets
  - collection: collection-b
    name: b1
    mount_path: /mnt/secrets
```
This is not passing the validation, right? What's wrong with this? Secrets are not colliding, do we want just to enforce a 1-to-1 mapping between the collection and the mount point?